File: assertion-roulette/java-junit/OrganizationUnitsControllerTests_Should_Create_OrganizationUnit_Async.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.springframework.http.ResponseEntity;
import java.util.UUID;
import java.util.Arrays;

public class OrganizationUnitsControllerTests {

    @Test
    public void shouldCreateOrganizationUnitAsync() throws Exception {
        var organizationUnitAppServiceMock = mock(IOrganizationUnitAppService.class);
        when(organizationUnitAppServiceMock.createAsync(any(CreateOrganizationUnitInput.class)))
            .thenReturn(new OrganizationUnitOutput(UUID.randomUUID(), 
                Arrays.asList(getTestUserOutput("test_user_for_insert_ou")), 
                Arrays.asList(getTestRoleOutput("test_role_for_insert_ou"))));

        var organizationUnitsController = new OrganizationUnitsController(organizationUnitAppServiceMock);
        ResponseEntity<OrganizationUnitOutput> actionResult = organizationUnitsController.postOrganizationUnits(new CreateOrganizationUnitInput());

        assertEquals(200, actionResult.getStatusCodeValue());
        var organizationUnitOutput = actionResult.getBody();
        assertNotNull(organizationUnitOutput);
        assertTrue(organizationUnitOutput.getSelectedUsers().size() > 0);
        assertTrue(organizationUnitOutput.getSelectedRoles().size() > 0);
    }
}

File: assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/OrganizationUnitsControllerTests_Should_Create_OrganizationUnit_Async.cs
using Xunit;

namespace BackOfficeBase.Tests.Web.Api
{
    public class OrganizationUnitsControllerTests
    {
        [Fact]
        public void Should_Create_OrganizationUnit_Async()
        {
            {
                var organizationUnitAppServiceMock = new Mock<IOrganizationUnitAppService>();
                organizationUnitAppServiceMock.Setup(x => x.CreateAsync(It.IsAny<CreateOrganizationUnitInput>())).ReturnsAsync(new OrganizationUnitOutput { Id = Guid.NewGuid(), SelectedUsers = new[] { GetTestUserOutput("test_user_for_insert_ou") }, SelectedRoles = new[] { GetTestRoleOutput("test_role_for_insert_ou") } });
                var organizationUnitsController = new OrganizationUnitsController(organizationUnitAppServiceMock.Object);
                var actionResult = await organizationUnitsController.PostOrganizationUnits(new CreateOrganizationUnitInput());
                var okObjectResult = Assert.IsType<OkObjectResult>(actionResult.Result);
                var organizationUnitOutput = Assert.IsType<OrganizationUnitOutput>(okObjectResult.Value);
                Assert.Equal((int)HttpStatusCode.OK, okObjectResult.StatusCode);
                Assert.True(organizationUnitOutput.SelectedUsers.Any());
                Assert.True(organizationUnitOutput.SelectedRoles.Any());
            }
        }
    }
}
